io/net/net_tools.py

```diff
@@ -0,0 +1,374 @@
+#!/usr/bin/env python
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2016 IBM
+# Author: Basheer K<basheer@linux.vnet.ibm.com>
+#
+# Based on code by  Hong Bo Peng <penghb@cn.ibm.com>
+# copyright: 2003, 2015 IBM Corp
+
+import os
+import re
+import shutil
+from avocado import Test
+from avocado import main
+from avocado.core import data_dir
+from avocado.utils import process, distro
+from avocado.utils.software_manager import SoftwareManager
+
+
+def install_dependencies():
+    """
+    To check and install  dependencies for the test
+    """
+    detected_distro = distro.detect()
+    sm = SoftwareManager()
+    if detected_distro.name == "SuSE":
+        net_tools = ("net-tools", "traceroute")
+    else:
+        net_tools = ("net-tools", "hostname", "traceroute")
+    for pkg in net_tools:
+        if not sm.check_installed(pkg) and not sm.install(pkg):
+            raise AssertionError("%s package is need to test" % pkg)
+
+
+class Hostname(Test):
+    """
+    hostname: set/get hostname by cmd. Check it with the expected result.
+    Test to verify the functionality of the hostname command
+    """
+
+    def setUp(self):
+        install_dependencies()
+        self.restore_hostname = False
+        # Get Hostname
+        output = process.run("hostname", verbose=False,
+                             env={"LANG": "C"})
+        if not output.stdout:
+            # set hostname if not set
+            if not process.system("hostname localhost.localdomain", sudo=True):
+                self.error("Failed to set hostname\n")
+
+        self.hostname = process.run("hostname", verbose=False,
+                                    env={"LANG": "C"}).stdout
```

Comment:
This second call is only needed when the hostname was updated. So how about:

```python
self.restore_hostname = False
hostname = process.system_output("hostname", verbose=False)
if not hostname:
    process.system("hostname localhost.localdomain", sudo=True)
    hostname = "localhost.localdomain"
self.hostname = hostname
```

PS: The `env={"LANG": "C"}` is only important for outputs where you match the output, it does not have to be everywhere.